build.sh

```diff
@@ -162,33 +162,83 @@ perl .pre-process-annotate-attributes.pl < $HTML_TEMP/source-expanded-1 > $HTML_
 perl .pre-process-tag-omission.pl < $HTML_TEMP/source-expanded-2 > $HTML_TEMP/source-whatwg-complete # this one could be merged
 mkdir $HTML_TEMP/wattsi-output
 
-if hash wattsi 2>/dev/null; then
-  # XXX wattsi --quiet awaits review https://github.com/whatwg/wattsi/pull/2
-  # In the mean time, wattsi --quiet will fail with "invalid arguments"
-  # unless you build from the wattsi pr/2 branch.
-  wattsi $($QUIET && echo "--quiet") \
-    $HTML_TEMP/source-whatwg-complete $HTML_TEMP/wattsi-output \
-    $HTML_CACHE/caniuse.json $HTML_CACHE/w3cbugs.csv
-else
-  $QUIET || echo
-  $QUIET || echo "Local wattsi is not present; trying the build server..."
-
-  HTTP_CODE=`curl $($VERBOSE && echo "-v") $($QUIET && echo "-s")\
-        http://ec2-52-88-42-163.us-west-2.compute.amazonaws.com/ \
-        --write-out "%{http_code}" \
-        --form source=@$HTML_TEMP/source-whatwg-complete \
-        --form caniuse=@$HTML_CACHE/caniuse.json \
-        --form w3cbugs=@$HTML_CACHE/w3cbugs.csv \
-        --output $HTML_TEMP/wattsi-output.zip`
-
-  if [ "$HTTP_CODE" != "200" ]; then
-      cat $HTML_TEMP/wattsi-output.zip
-      rm -f $HTML_TEMP/wattsi-output.zip
-      exit 22
+function reRunWattsiOrReportErrors {
+  # If we end up here it means that wattsi exited non-zero, probably with
+  # one or more error messages that include line and column numbers; e.g.:
+  #     Parse Error:(748,59) unexpected end tag
+  # But in that case, the line numbers that wattsi reports are for lines in
+  # the copy of the source produced by running the original through the
+  # pre-processer steps above. Therefore, the reported line numbers may be
+  # wrong, so here we re-run wattsi against the original source to get the
+  # correct line numbers.
+  if [[ "$1" != $HTML_SOURCE/source ]]; then
+    # we get here if we've done a first pass through wattsi and errors were
+    # reported; so we re-run now to get the right line numbers
+    rm -rf $HTML_TEMP/wattsi-output && mkdir $HTML_TEMP/wattsi-output
+    runWattsi $HTML_SOURCE/source
+  else
+    # we get after we've done a second pass through wattsi; so it's now
+    # time to emit any error messages wattsi reported
+    if [[ -f $HTML_TEMP/wattsi-first-pass-output.txt \
+      && -n "$(cat $HTML_TEMP/wattsi-first-pass-output.txt)" ]]; then
+      cat $HTML_TEMP/wattsi-first-pass-output.txt | grep -v '^$' # trim blank lines
```

Comment:
I don't quite understand the logic here. Shouldn't it be, "if wattsi-first-pass-output exists but wattsi-output is empty, emit wattsi-first-pass-output"?

Similarly, the below would be "if wattsi-output exists and is nonempty, output it" whereas right now it's "if wattsi-output exists and is nonempty, emit wattsi-first-pass-output".

I would suggest maybe the following if chain:

```bash
if isNonemptyFile $HTML_TEMP/wattsi-output.txt
  cat $HTML_TEMP/wattsi-output.txt
else
  cat $HTML_TEMP/wattsi-first-pass-output.txt
fi
```

?

Let me try prototyping...